src/traceViewer/web/ui/timeline.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2017 Google Inc. All rights reserved.
+ * Modifications copyright (c) Microsoft Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { ContextEntry } from '../../traceModel';
+import './timeline.css';
+import { FilmStrip } from './filmStrip';
+import { Boundaries } from '../components/geometry';
+import * as React from 'react';
+import { useMeasure } from './helpers';
+
+export const Timeline: React.FunctionComponent<{
+  context: ContextEntry,
+  boundaries: Boundaries,
+}> = ({ context, boundaries }) => {
+  const [measure, ref] = useMeasure<HTMLDivElement>();
+  const [previewX, setPreviewX] = React.useState<number | undefined>();
+
+  const offsets = React.useMemo(() => {
+    return calculateDividerOffsets(measure.width, boundaries);
+  }, [measure.width, boundaries]);
+  const actionEntries = React.useMemo(() => {
+    const actions = [];
+    for (const page of context.pages)
+      actions.push(...page.actions);
+    return actions;
+  }, [context]);
+  const actionTimes = React.useMemo(() => {
+    return actionEntries.map(entry => {
+      return {
+        action: entry.action,
+        actionId: entry.actionId,
+        left: timeToPercent(measure.width, boundaries, entry.action.startTime!),
+        right: timeToPercent(measure.width, boundaries, entry.action.endTime!),
+      };
+    });
+  }, [actionEntries, boundaries, measure.width]);
+
+  const onMouseMove = (event: React.MouseEvent) => {
+    if (ref.current)
+      setPreviewX(event.clientX - ref.current.getBoundingClientRect().left);
+  };
+  const onMouseLeave = () => {
+    setPreviewX(undefined);
+  };
+
+  return <div ref={ref} className='timeline-view' onMouseMove={onMouseMove} onMouseLeave={onMouseLeave}>
```

Comment:
```suggestion
  return <div ref={ref} className='timeline-view' onMouseMove={onMouseMove} onMouseOver={onMouseMove} onMouseLeave={onMouseLeave}>
```
Iirc different browsers behave differently here. I don't think you are guaranteed a mousemove depending on how the mouse enters.